src/contracts/core/DelegationManager.sol

```diff
@@ -533,20 +521,21 @@ contract DelegationManager is
         uint256 addedShares,
         uint64 maxMagnitude
     ) internal {
-        uint256 slashingFactor = _getSlashingFactor(staker, strategy, maxMagnitude);
-
         // Ensure that the operator has not been fully slashed for a strategy
-        // and that the staker has not been fully slashed if its the beaconChainStrategy
+        // and that the staker has not been fully slashed if it is the beaconChainStrategy
+        uint256 slashingFactor = _getSlashingFactor(staker, strategy, maxMagnitude);
         require(slashingFactor != 0, FullySlashed());
 
-        // Increment operator shares
-        operatorShares[operator][strategy] += addedShares;
-        emit OperatorSharesIncreased(operator, staker, strategy, addedShares);
-
         // Update the staker's depositScalingFactor
         DepositScalingFactor storage dsf = _depositScalingFactor[staker][strategy];
         dsf.update(existingDepositShares, addedShares, slashingFactor);
         emit DepositScalingFactorUpdated(staker, strategy, dsf.scalingFactor());
```

Comment:
Noting: The DSF won't update because staker is undelegated => maxMagnitude is 0 => DSF is always equal to WAD on each new deposit. However, if it is the beaconChainStrategy and the BCSF is non-WAD then this will update.

Reply:
I'll add a comment but I'd rather keep the code path without a branch for the BCSF. Good note, ty

Reply:
[34ba24b](https://github.com/Layr-Labs/eigenlayer-contracts/pull/876/commits/34ba24be2ffa25c33f8c45803a56bb9377010a31)